### ISSUE TITLE
Missing required mutation input parameter is not detected

### ISSUE THREAD

**Issue by BaloghTamas** [apollographql/apollo-kotlin#3891]
**Summary**
When a mutation has a required parameter, but it is not supplied, it is not detected at compile time, but fails with `HTTP 400 - BAD REQUEST` in runtime, I think this should be detected at compile time

**Version**
2.5.11

**Description**
This should be reproducable with the following mutation:

**schema**
```
type MySettings {
  edit(input: MySettingEditInput!): MySettingEditResult
}

input MySettingsEditInput {
  someOptionalParameter: String
}

type MySettingEditResult {
  someResult: String
}
```

**mutation that compiles but fails in runtime**
```
mutation UpdateMySettings {
    edit {
      someResult
 }
```

**mutation that compiles and works in runtime**
```
mutation UpdateMySettings {
    edit(input: {}) {
      someResult
 }
```

**Comment by martinbonnin** [comment]
Thanks for sending this! [A fix will be available in 3.1.2](https://github.com/apollographql/apollo-kotlin/pull/3893). 

Unless you have strong objections, I'm not planning to backport this to 2.x since there is always the workaround of adding the argument manually?

**Comment by BaloghTamas** [comment]
> Unless you have strong objections, I'm not planning to backport this to 2.x since there is always the workaround of adding the a argument manually?

Yes sure, we had applied that workaround for now :+1:
